ptpython/python_input.py

```diff
@@ -379,10 +380,17 @@ def __init__(
             extra_toolbars=self._extra_toolbars,
         )
 
-        self.app = self._create_application(input, output)
+        # Create an app if requested. If not, the global get_app() is returned
+        # for self.app via property getter.
+        if create_app:
```

Comment:
I think the ``app`` creation can be moved to a method, or to the ``PythonRepl`` entirely.

Reply:
Also, the following parameters then have to move this to method or the place where the ``App`` is created:

- vi_mode
- color_depth
- input
- output
- extra_key_bindings

Reply:
edit: thinking more about it, it's not that simple. The trouble with ptpython right now is that the config file has access to the ``PythonRepl/PythonInput`` object. All configuration is done by manipulating that object. So any change here is potentially breaking peoples configuration. I somewhat regret this now, but this was a simple way back then to allow people to configure about anything.

Maybe for now it's fine. It would be good to document the meaning of ``create_app`` though, because this one is not obvious.

Reply:
Sounds good, I added a bit about `create_app` to the PythonInput doc string for now.